# 1 задача
- Скачать файл в уроке
- Считать данные с помощью pandas
- Вывести на экран первые 5 строк
- Посмотреть на описание признаков и на их содержание

In [72]:
import pandas as pd
import numpy as np

In [75]:
df = pd.read_csv('kc_house_data.csv', encoding='windows-1251', index_col='id', parse_dates=['date'], dayfirst=False)

In [76]:
df.head()

date     price  bedrooms  bathrooms  sqft_living  sqft_lot  \
id                                                                            
7129300520 2014-10-13  221900.0         3       1.00         1180      5650   
6414100192 2014-12-09  538000.0         3       2.25         2570      7242   
5631500400 2015-02-25  180000.0         2       1.00          770     10000   
2487200875 2014-12-09  604000.0         4       3.00         1960      5000   
1954400510 2015-02-18  510000.0         3       2.00         1680      8080   

            floors  waterfront  view  condition  grade  sqft_above  \
id                                                                   
7129300520     1.0           0     0          3      7        1180   
6414100192     2.0           0     0          3      7        2170   
5631500400     1.0           0     0          3      6         770   
2487200875     1.0           0     0          5      7        1050   
1954400510     1.0           0     0          3      8        1680   

            sqft_basement  yr_built  yr_renovated  zipcode      lat     long  \
id                                                                             
7129300520              0      1955             0    98178  47.5112 -122.257   
6414100192            400      1951          1991    98125  47.7210 -122.319   
5631500400              0      1933             0    98028  47.7379 -122.233   
2487200875            910      1965             0    98136  47.5208 -122.393   
1954400510              0      1987             0    98074  47.6168 -122.045   

            sqft_living15  sqft_lot15  
id                                     
7129300520           1340        5650  
6414100192           1690        7639  
5631500400           2720        8062  
2487200875           1360        5000  
1954400510           1800        7503

#  2 задача
## Проведите первичный анализ данных
- Изучите типы данных
- Пишите выводы

In [78]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 21613 entries, 7129300520 to 1523300157
Data columns (total 20 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   date           21613 non-null  datetime64[ns]
 1   price          21613 non-null  float64       
 2   bedrooms       21613 non-null  int64         
 3   bathrooms      21613 non-null  float64       
 4   sqft_living    21613 non-null  int64         
 5   sqft_lot       21613 non-null  int64         
 6   floors         21613 non-null  float64       
 7   waterfront     21613 non-null  int64         
 8   view           21613 non-null  int64         
 9   condition      21613 non-null  int64         
 10  grade          21613 non-null  int64         
 11  sqft_above     21613 non-null  int64         
 12  sqft_basement  21613 non-null  int64         
 13  yr_built       21613 non-null  int64         
 14  yr_renovated   21613 non-null  int64         
 15  zipco

In [80]:
df.shape

(21613, 20)

## Вывод
- 21613 записей
- 21 столбец
- Есть типы: вещественные, целые, строковые

- Найдите количество пропущенных ячеек в данных

In [81]:
df.isna().sum()

date             0
price            0
bedrooms         0
bathrooms        0
sqft_living      0
sqft_lot         0
floors           0
waterfront       0
view             0
condition        0
grade            0
sqft_above       0
sqft_basement    0
yr_built         0
yr_renovated     0
zipcode          0
lat              0
long             0
sqft_living15    0
sqft_lot15       0
dtype: int64

## Вывод
- Пропущенных ячеек нет

- Посчитайте основные статистики по всем признакам и поизучайте их

In [36]:
df.describe()

price      bedrooms     bathrooms   sqft_living      sqft_lot  \
count  2.161300e+04  21613.000000  21613.000000  21613.000000  2.161300e+04   
mean   5.400881e+05      3.370842      2.114757   2079.899736  1.510697e+04   
std    3.671272e+05      0.930062      0.770163    918.440897  4.142051e+04   
min    7.500000e+04      0.000000      0.000000    290.000000  5.200000e+02   
25%    3.219500e+05      3.000000      1.750000   1427.000000  5.040000e+03   
50%    4.500000e+05      3.000000      2.250000   1910.000000  7.618000e+03   
75%    6.450000e+05      4.000000      2.500000   2550.000000  1.068800e+04   
max    7.700000e+06     33.000000      8.000000  13540.000000  1.651359e+06   

             floors    waterfront          view     condition         grade  \
count  21613.000000  21613.000000  21613.000000  21613.000000  21613.000000   
mean       1.494309      0.007542      0.234303      3.409430      7.656873   
std        0.539989      0.086517      0.766318      0.650743      1.175459   
min        1.000000      0.000000      0.000000      1.000000      1.000000   
25%        1.000000      0.000000      0.000000      3.000000      7.000000   
50%        1.500000      0.000000      0.000000      3.000000      7.000000   
75%        2.000000      0.000000      0.000000      4.000000      8.000000   
max        3.500000      1.000000      4.000000      5.000000     13.000000   

         sqft_above  sqft_basement      yr_built  yr_renovated       zipcode  \
count  21613.000000   21613.000000  21613.000000  21613.000000  21613.000000   
mean    1788.390691     291.509045   1971.005136     84.402258  98077.939805   
std      828.090978     442.575043     29.373411    401.679240     53.505026   
min      290.000000       0.000000   1900.000000      0.000000  98001.000000   
25%     1190.000000       0.000000   1951.000000      0.000000  98033.000000   
50%     1560.000000       0.000000   1975.000000      0.000000  98065.000000   
75%     2210.000000     560.000000   1997.000000      0.000000  98118.000000   
max     9410.000000    4820.000000   2015.000000   2015.000000  98199.000000   

                lat          long  sqft_living15     sqft_lot15  
count  21613.000000  21613.000000   21613.000000   21613.000000  
mean      47.560053   -122.213896    1986.552492   12768.455652  
std        0.138564      0.140828     685.391304   27304.179631  
min       47.155900   -122.519000     399.000000     651.000000  
25%       47.471000   -122.328000    1490.000000    5100.000000  
50%       47.571800   -122.230000    1840.000000    7620.000000  
75%       47.678000   -122.125000    2360.000000   10083.000000  
max       47.777600   -121.315000    6210.000000  871200.000000

# 3 задача
## Ответьте на несколько вопросов
- 3.1 В каком диапазоне изменяются стоимости недвижимости?


In [84]:
f"min {df['price'].min()}", f"max {df['price'].max()}"

('min 75000.0', 'max 7700000.0')

- 3.2 Какую долю в среднем занимают жилая площадь от всей площади по всем домам?

In [86]:
sqrt_liv_sum = df['sqft_living'].mean()
sqrt_lot_sum = df['sqft_lot'].mean()
rez = round(sqrt_liv_sum / sqrt_lot_sum * 100, 2)
rez

13.77

- 3.3 Как много домов с разными этажами в данных?

In [40]:
df['floors'].value_counts()

1.0    10680
2.0     8241
1.5     1910
3.0      613
2.5      161
3.5        8
Name: floors, dtype: int64

- 3.4 Насколько хорошие состояния у домов в данных?

In [41]:
df['condition'].value_counts(normalize=True)*100,'%'

(3    64.919262
 4    26.275852
 5     7.870263
 2     0.795817
 1     0.138805
 Name: condition, dtype: float64,
 '%')

- 3.5 Найдите года, когда построили первый дом, когда построили последний дом в данных?

In [87]:
f"first {df['yr_built'].min()}", f"last {df['yr_built'].max()}"

('first 1900', 'last 2015')

# 4 задача
## Ответьте на несколько вопросов

- 4.1 Сколько в среднем стоят дома, у которых 2 спальни?

In [46]:
df[df['bedrooms'] == 2]['price'].mean()

401372.681884058

- 4.2 Какая в среднем общая площадь домов, у которых стоимость больше 600 000? 

In [88]:
df[df['price'] > 600000]['sqft_lot'].mean()

20442.524776214832

- 4.3 Как много домов коснулся ремонт? 

In [48]:
df[(df['yr_renovated'] != 0)].shape[0]

914

- 4.4 Насколько в среднем стоимость домов с оценкой grade домов выше 10 отличается от стоимости домов с оценкой grade меньше 4?

In [50]:
df[(df['grade'] > 10)]["price"].mean() - df[(df['grade'] < 4)]["price"].mean()

1488885.1175298805

# 5 задача
## Ответьте на несколько вопросов
- 5.1 Выберите дом клиенту
- Клиент хочет дом с видом на набережную, как минимум с тремя ванными и с подвалом. Сколько вариантов есть у клиента?
- 5.3 Выберите дом клиенту
- Клиент хочет дом без подвала, с двумя этажами, стоимостью до 150000. Какая оценка по состоянию у таких домов в среднем?

In [59]:
df[(df['waterfront'] == 1) & (df['bathrooms'] >= 3) & (df['sqft_basement'] != 0)].shape[0]

41

- 5.2 Выберите дом клиенту
- Клиент хочет дом либо с очень красивым видом из окна, либо с видом на набережную, в очень хорошем состоянии и год постройки не меньше 1980 года. В какой ценовом диапазоне будут дома?

In [89]:
client_list = df[((df['view'] == 4) | (df['waterfront'] == 1)) & (df['condition'] == 5) & (df['yr_built'] >= 1980)]['price']
client_list.min(),client_list.max()

(1295000.0, 3000000.0)

- 5.3 Выберите дом клиенту
- Клиент хочет дом без подвала, с двумя этажами, стоимостью до 150000. Какая оценка по состоянию у таких домов в среднем?

In [104]:
round(df[(df['sqft_basement'] == 0) & (df['floors'] == 2) & (df['price'] < 150000)]['condition'].mean(), 2)

2.83